src/shared/components/Searchbar/Searchbar.tsx

```diff
@@ -1,87 +1,111 @@
 import React, { useState } from 'react'
+import { CSSTransition } from 'react-transition-group'
 
 import { useMediaMatch } from '@/hooks/useMediaMatch'
-import { SvgGlyphClose, SvgGlyphSearch } from '@/shared/icons'
+import { Button } from '@/shared/components/Button'
+import { ShortcutIndicator } from '@/shared/components/ShortcutIndicator'
+import { SvgGlyphChevronLeft, SvgGlyphClose, SvgGlyphSearch } from '@/shared/icons'
 
+import { SearchBox } from './SearchBox'
+import { SearchHelper } from './Searchbar.style'
 import { CancelButton, Container, Input, SearchButton, StyledSvgOutlineSearch } from './Searchbar.style'
 
 type SearchbarProps = {
-  value: string
+  value: string | null
   onCancel?: () => void
   showCancelButton?: boolean
   controlled?: boolean
   onClick?: () => void
+  hasFocus: boolean
+  onClose: () => void
 } & React.DetailedHTMLProps<React.HTMLAttributes<HTMLInputElement>, HTMLInputElement>
-export const Searchbar: React.FC<SearchbarProps> = ({
-  placeholder,
-  onChange,
-  onFocus,
-  onCancel,
-  showCancelButton = false,
-  controlled = false,
-  value: externalValue,
-  onBlur,
-  onSubmit,
-  className,
-  onClick,
-  ...htmlProps
-}) => {
-  const [value, setValue] = useState('')
-  const mdMatch = useMediaMatch('md')
+export const Searchbar = React.forwardRef<HTMLDivElement, SearchbarProps>(
+  (
+    {
+      placeholder,
+      onChange,
+      onFocus,
+      onCancel,
+      controlled = false,
+      value: externalValue,
```

Comment:
Do we still need both the controlled and uncontrolled states? I think we only use one of them so maybe let's just remove the logic for the other one?